test/source/mock/google-api-mock.ts

```diff
@@ -190,9 +190,28 @@ const parseMultipartDataAsMimeMsg = async (multipartData: string): Promise<Parse
   return parsed;
 };
 
-const validateMimeMsg = (acct: string, mimeMsg: ParsedMail, threadId?: string) => {
-  if (threadId && !new Data(acct).getThreads().find(t => t.id === threadId)) {
-    throw new HttpClientErr('The thread you are replying to not found', 404);
+const validateMimeMsg = async (acct: string, mimeMsg: ParsedMail, threadId?: string) => {
+  if (threadId) {
+    const messages = new Data(acct).getMessagesByThread(threadId);
+    if (!messages || !messages.length) {
+      throw new HttpClientErr('The thread you are replying to not found', 404);
+    }
+    const inReplyToMessageId = mimeMsg.headers.get('in-reply-to') ? mimeMsg.headers.get('in-reply-to')!.toString() : '';
+    if (inReplyToMessageId) {
```

Comment:
If there is `threadId` but no `inReplyToMessageId`, it should fail - very loudly - that's the most important thing to test.

Reply:
So, if there is a `threadId` it's a reply message anyway, yes?

Reply:
Yes, from the perspective of the sender. It will be appended to convo on recipient side.

From the perspective of the recipient, that depends on the email headers.